src/std/env.ab

```diff
@@ -76,52 +89,59 @@ pub fun is_root(): Bool {
     return false
 }
 
+/// ?
 pub fun printf(format: Text, args: [Text] = [""]): Null {
     unsafe ${nameof args}=("{format}" "\$\{{nameof args}[@]}")$
     unsafe $printf "\$\{{nameof args}[@]}"$
 }
 
+/// ?
 pub fun printf_escape(text: Text): Text {
     return unsafe $echo \${nameof text} | sed -e 's/\\\\/\\\\\\\\/g' -e "s/%/%%/g"$
 }
 
+/// Print a string with formatting options
 pub fun text_shell(message: Text, style: Num, fg: Num, bg: Num): Text {
     return "\x1b[{style};{fg};{bg}m{printf_escape(message)}\x1b[0m"
 }
```

Comment:
not printing but "prepare" this function return a string can be passed to printf function.
`message` must be escaped before send to printf function
if you want to print `\version` in bold `\` must be escaped